### PR TITLE
Allow dotenv files from `--environment`

### PR DESCRIPTION
**Pull request by Lorquas**
Pull Request Checklist

* [x] implement the feature
* [x] adjust docstring
* [ ] mention the version

According to the man/help both yaml and dotenv are supported:
```
$ tmt --version
tmt version: 1.49.0
$ tmt run --help 
...
  -e, --environment KEY=VALUE|@FILE
                                  Set environment variable. Can be specified
                                  multiple times. The "@" prefix marks a file
                                  to load (yaml or dotenv formats
                                  supported).
```
however it only accepts YAML:
```
$ echo MYVAR=foobar > test.env
$ tmt -vvv run -a --environment @test.env --dry
/var/tmp/tmt/run-008

Failed to extract variables from YAML format, YAML defining variables must be a dictionary.
$ echo $?
2
```

This PR fixes this bug.

### CODE REVIEW

**Review by therazix — APPROVED** [review]
LGTM. Works as expected.

**Review by tcornell-bus — CHANGES_REQUESTED** [review]
A quick test added would be really helpful to make sure .env files stay supported. I think there is already tests for .yaml here `tests/core/env`. Maybe a simple .env test can be added there as well?

Otherwise works as expect for me!

**Comment by happz** [comment]
Indeed, a test would be nice. @Lorquas would you be able to extend `/tests/core/env` to cover also dotenv files?

File: tests/core/env/test.sh [hunk 1]
```diff
@@ -56,6 +56,20 @@ rlJournalStart
...
+                        plan --name $plan \
+                        test --name $test"
+                    rlAssertGrep '>>>O0<<<' $rlRun_LOG
+                done
+            done
+        rlPhaseEnd
+
         rlPhaseStartTest "Empty environment file ($execute)"
             rlRun -s "tmt run -r -e @empty.yaml"
             rlAssertGrep "warn: Empty environment file" $rlRun_LOG
```
Comment (thrix) [inline tests/core/env/test.sh]: @Lorquas also please add a test for empty environment file

Comment (Lorquas) [reply tests/core/env/test.sh]: done!

File: tests/core/env/test.sh [hunk 2]
```diff
@@ -56,10 +56,29 @@ rlJournalStart
...
+                    rlRun -s "tmt run -avvvr -e @vars.env \
+                        execute --how $execute \
+                        plan --name $plan \
+                        test --name $test"
+                    rlAssertGrep '>>>O0<<<' $rlRun_LOG
+                done
+            done
+        rlPhaseEnd
+
         rlPhaseStartTest "Empty environment file ($execute)"
```
Comment (thrix) [inline tests/core/env/test.sh]:
```suggestion
        rlPhaseStartTest "Empty YAML environment file ($execute)"
```

File: tests/core/env/test.sh [hunk 3]
```diff
@@ -56,10 +56,29 @@ rlJournalStart
...
             rlRun -s "tmt run -r -e @empty.yaml"
             rlAssertGrep "warn: Empty environment file" $rlRun_LOG
         rlPhaseEnd
+
+        rlPhaseStartTest "Empty environment file ($execute)"
```
Comment (thrix) [inline tests/core/env/test.sh]:
```suggestion
        rlPhaseStartTest "Empty DOTENV environment file ($execute)"
```
